pkg/controller/changetierrequest/changetierrequest_controller.go

```diff
@@ -144,14 +145,16 @@ func (r *ReconcileChangeTierRequest) changeTier(log logr.Logger, changeTierReque
 	namespaces := make([]toolchainv1alpha1.NSTemplateSetNamespace, len(nsTemplateTier.Spec.Namespaces))
 	for i, ns := range nsTemplateTier.Spec.Namespaces {
 		namespaces[i] = toolchainv1alpha1.NSTemplateSetNamespace{
-			Type:     ns.Type,
-			Revision: ns.Revision,
+			Type:        ns.Type,
+			Revision:    ns.Revision,
+			TemplateRef: nstemplatetiers.NewTierTemplateName(changeTierRequest.Spec.TierName, ns.Type, ns.Revision), // link to the TierTemplate resource, whose name is: `<tierName>-<nsType>-<revision>`,
 		}
 	}
 	var clusterResources *toolchainv1alpha1.NSTemplateSetClusterResources
 	if nsTemplateTier.Spec.ClusterResources != nil {
 		clusterResources = &toolchainv1alpha1.NSTemplateSetClusterResources{
 			Revision: nsTemplateTier.Spec.ClusterResources.Revision,
+			//TemplateRef: nstemplatetiers.NewTierTemplateName(changeTierRequest.Spec.TierName, nstemplatetiers.ClusterResources, nsTemplateTier.Spec.ClusterResources.Revision), // link to the TierTemplate resource, whose name is: `<tierName>-<nsType>-<revision>`,
```

Comment:
you need to set the TemplateRef here:
```suggestion
			TemplateRef: nsTemplateTier.Spec.ClusterResources.TemplateRef,
```